includes/api/class-wc-rest-product-variations-controller.php

```diff
@@ -248,11 +249,17 @@ protected function prepare_object_for_database( $request, $creating = false ) {
 
 		$variation->set_parent_id( absint( $request['product_id'] ) );
 
-		// Status.
+		// TODO Fix/Remove in future API version bump. This controls the wrong thing.
+		// https://github.com/woocommerce/woocommerce/issues/15215
 		if ( isset( $request['visible'] ) ) {
 			$variation->set_status( false === $request['visible'] ? 'private' : 'publish' );
 		}
 
+		// Status.
```

Comment:
I feel like most clients are going to target a particular WooCommerce version anyway. However -- if we are bumping the version I don't think we should do it in this PR. It should be a separate PR since we are also adding the locations endpoints.